CMakeLists.txt

```diff
@@ -382,6 +382,11 @@ find_path(
 
 if(PG_SOURCE_DIR)
   message(STATUS "Found PostgreSQL source in ${PG_SOURCE_DIR}")
+  # Add the PostgreSQL source dir include directories to the build system
+  # includes BEFORE the installed PG include files. This will allow the LSP
+  # (e.g., clangd) to navigate to the PostgreSQL source instead of the install
+  # path directory that only has the headers.
+  include_directories(BEFORE SYSTEM ${PG_SOURCE_DIR}/src/include)
```

Comment:
We have `include_directories(SYSTEM ${PG_INCLUDEDIR_SERVER})` in `build-defs.cmake` already, I wonder why it doesn't work for clangd.

Reply:
Using includes from source might be slightly confusing, because Timescale is compiled against the `${PG_INCLUDEDIR_SERVER}`, which doesn't necessarily match the source includes. E.g. if you're using several out-of-source builds for Postgres with different configurations, the `pg_config.h` will be different.

Reply:
> We have `include_directories(SYSTEM ${PG_INCLUDEDIR_SERVER})` in `build-defs.cmake` already, I wonder why it doesn't work for clangd.

`include_directories(SYSTEM ${PG_INCLUDEDIR_SERVER})` points to a different directory.

Good point about the `pg_config.h`. We could make this an optional override on the command line to protect against it.

I also tried using `CMAKE_C_STANDARD_INCLUDE_DIRECTORIES` on the command line but that always puts the PG source include _after_ the installed header include directory, which won't help.

Reply:
The problem is that your IDE can't go to definition from the installed headers, right? That's why you want the ones from source. Personally I just added the entire PG source into the IDE project and it works, but I'm using QT Creator, not sure how this translates to clangd.

Reply:
Yeah, that's the problem. I am using Emacs + clangd, although I think it is a problem for all IDEs that use language server protocol (LSP) based on `compile_commands.json` unless they build their own index using some other method.

Reply:
@akuzm Added an option (default OFF) to enable this behavior. Now it won't interfere with out-of-source builds.